app/controllers/api/v1/simulator_controller.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+class Api::V1::SimulatorController < Api::V1::BaseController
+  include SimulatorHelper
+  include ActionView::Helpers::SanitizeHelper
+
+  before_action :authenticate_user!, only: %i[create update edit]
+  before_action :set_project, only: %i[data edit]
+  before_action :set_user_project, only: %i[update]
+  before_action :check_view_access, only: %i[data]
+  before_action :check_edit_access, only: %i[edit update]
+  # skip_before_action :verify_authenticity_token, only: %i[data create update verilogcv]
+
+  def self.policy_class
+    ProjectPolicy
+  end
+
+  # GET api/v1/simulator/:id/edit
+  def edit
```

Comment:
I am not sure with method name `edit`, it meant to update something but here it is return a `user`, can we give the method name as `show`, or something, WDYT? @Arnabdaz

Reply:
@vedant-jain03 I was also confused about naming it as `edit` but then `show` is currently used for viewing the circuit ... so people might not have access to edit a circuit but can have access to view it, if the circuit is public. 
what do you think about naming it as `edit_access` or `check_access` or `check_edit_access`